### PR TITLE
[Docs] Specify Memory table engine usage on cloud

### PR DESCRIPTION
**Pull request by justindeguzman**
### Changelog category (leave one):
- Documentation (changelog entry is not required)

### Changelog entry (a user-readable short description of the changes that goes to CHANGELOG.md):
Specify Memory table engine usage on cloud

### CODE REVIEW

**Comment by robot-ch-test-poll4** [comment]
*This is an automated comment for commit b9bb04ebf03f6677d6f81ac628ea94ca3b9b43bd with description of existing statuses. It's updated for the latest CI running*

[❌ Click here](https://s3.amazonaws.com/clickhouse-test-reports/59609/b9bb04ebf03f6677d6f81ac628ea94ca3b9b43bd/ci_running.html) to open a full report in a separate page

<table>
<thead><tr><th>Check name</th><th>Description</th><th>Status</th></tr></thead>
<tbody>
<tr><td>PR Check</td><td>There's no description for the check yet, please add it to tests/ci/ci_config.py:CHECK_DESCRIPTIONS</td><td>❌ failure</td></tr>
<tbody>
</table>

**Review by melvynator — COMMENTED** [review]
When using the Memory table engine on ClickHouse Cloud data are not replicated across all the nodes, this is by design. To use it correctly you need to do all your operations in the same session or using a client with a sticky connection (like the clickhouse-client), it will guarantee that all the queries are rooted to the same node.

File: docs/en/engines/table-engines/special/memory.md [hunk 1]
```diff
@@ -6,6 +6,10 @@ sidebar_label:  Memory
 
 # Memory Table Engine
 
+:::note
+When using the Memory table engine on ClickHouse Cloud, you must use a client that uses TCP (such as [clickhouse-client](/en/interfaces/cli)) or a native interface, and not one that uses HTTP (such as [clickhouse-connect](/en/integrations/python)). If you use HTTP, all queries must be submitted and executed at once using a multi-statement query.
```
Comment (alexey-milovidov) [inline docs/en/engines/table-engines/special/memory.md:10]: The second statement contradicts the word "must" in the first statement.

**Review by alexey-milovidov — CHANGES_REQUESTED** [review]
.

File: docs/en/engines/table-engines/special/memory.md [hunk 2]
```diff
@@ -6,6 +6,13 @@ sidebar_label:  Memory
 
 # Memory Table Engine
 
+:::note
+When using the Memory table engine on ClickHouse Cloud, data is not replicated across all nodes (by design). To guarantee that all queries are routed to the same node and that the Memory table engine works as expected, you can do one of the following:
+- Execute all operations in the same session
+- Use a client that uses TCP or the native interface (which enables support for sticky connections) such as [clickhouse-client](/en/interfaces/cli)
+- Submit and execute all queries at once using a multi-statement query (required with clients using the HTTP interface such as [clickhouse-connect](/en/integrations/python))
```
Comment (alexey-milovidov) [inline docs/en/engines/table-engines/special/memory.md:13]: I'm not sure, are multi-statements supported for HTTP? I thought the opposite.

Comment (justindeguzman) [reply docs/en/engines/table-engines/special/memory.md:13]: You're right, I misinterpreted. I can remove the last bullet point.

**Review by alexey-milovidov — APPROVED** [review]

**Pull request merged**